shell/platform/android/io/flutter/plugin/common/MethodCall.java

```diff
@@ -0,0 +1,29 @@
+package io.flutter.plugin.common;
+
+import java.util.Objects;
+
+/**
+ * Command object representing a method call on a {@link FlutterChannel}.
+ */
+public final class MethodCall {
+    /**
+     * The name of the called method.
+     */
+    public final String method;
+
+    /**
+     * Arguments for the call.
+     */
+    public final Object arguments;
```

Comment:
> Do we have a good reason for enforcing arguments to be held in an array (or list or map) across all methods invoked by the JSON codec?

Nope.

> Separation of concerns if often helpful, but splitting codecs for messages and method calls (and event streams?) also has a price.

What is the price?

> Do we then also want multiple channel concepts (or does it even make sense then), or do we configure each channel with two or three different codecs?

I'd recommend having two types of channels.  One that transports messages and another that transports method calls.